internal/converter/internal/otelcolconvert/converter_vcenterreceiver.go

```diff
@@ -0,0 +1,83 @@
+package otelcolconvert
+
+import (
+	"fmt"
+
+	"github.com/grafana/agent/internal/component/otelcol"
+	"github.com/grafana/agent/internal/component/otelcol/receiver/vcenter"
+	"github.com/grafana/agent/internal/converter/diag"
+	"github.com/grafana/agent/internal/converter/internal/common"
+	"github.com/grafana/river/rivertypes"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver"
+	"go.opentelemetry.io/collector/component"
+)
+
+func init() {
+	converters = append(converters, vcenterReceiverConverter{})
+}
+
+type vcenterReceiverConverter struct{}
+
+func (vcenterReceiverConverter) Factory() component.Factory { return vcenterreceiver.NewFactory() }
+
+func (vcenterReceiverConverter) InputComponentName() string { return "" }
+
+func (vcenterReceiverConverter) ConvertAndAppend(state *state, id component.InstanceID, cfg component.Config) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	label := state.FlowComponentLabel()
+
+	args := toVcenterReceiver(state, id, cfg.(*vcenterreceiver.Config))
+	block := common.NewBlockWithOverride([]string{"otelcol", "receiver", "vcenter"}, label, args)
+
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Converted %s into %s", stringifyInstanceID(id), stringifyBlock(block)),
+	)
+
+	state.Body().AppendBlock(block)
+	return diags
+}
+
+func toVcenterReceiver(state *state, id component.InstanceID, cfg *vcenterreceiver.Config) *vcenter.Arguments {
+	var (
+		nextMetrics = state.Next(id, component.DataTypeMetrics)
+		nextTraces  = state.Next(id, component.DataTypeTraces)
+	)
+
+	return &vcenter.Arguments{
+		Endpoint: cfg.Endpoint,
+		Username: cfg.Username,
+		Password: rivertypes.Secret(cfg.Password),
+
```

Comment:
It looks like we are missing the conversion of `MetricsBuilderConfig` here.

Reply:
I've added the converter but it looks like the defaults for agent's `vcenter` doesn't seem to be in sync with upstream, hence the full-blown River converted result.

These blocks with `enabled = true` should be omitted, right? Just wanna make sure that I understand the concept of `upstream syncing` correctly here

Reply:
Good catch - I researched it and it made sense to commit the change I tested. The way nested structs were being handled in the component led to this weird behavior and it should be all set now. There is a slight divergence in defaults but it is not so bad anymore.